### PR TITLE
Update CONTRIBUTING.md code structure section

### LINKED ISSUE #2592: Update CONTRIBUTING.md code structure section
Our CONTRIBUTING.md file contains outdated information. The section describing the code structure needs to be update to reflect the current code strucuture.

### PR DESCRIPTION
**Pull request by elomscansio**
The CONTRIBUTING.md file contains outdated information in the section describing the code structure which I updated to reflect the current code strucuture.

### CODE REVIEW

**Review by cathay4t — DISMISSED** [review]

**Review by ffmancera — CHANGES_REQUESTED** [review]
While I agree with removing these items, the PR is incomplete. There are new directories like `src/`, `src/rust/.. and others. It would be really useful to have them documented.

If you have questions about them, feel free to ask us.

**Review by cathay4t — CHANGES_REQUESTED** [review]
Besides removing outdated lines, please also include doc for new code layout:

 * `rust/src/lib` for rust crate.
 * `rust/src/cli` for `nmstatectl` command line tool.
 * etc

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -24,20 +24,18 @@ serving the tests of Nmstate.
...
+- `./rust/src/lib/ifaces/` Contains the classes including the attributes of the 
 interfaces.
  
-- `./libnmstate/nispor/` Contains the nmstate nispor plugin, currently for 
+- `./rust/src/lib/nispor/` Contains the nmstate nispor plugin, currently for 
 querying kernel network status. 
 
-- `./libnmstate/nm/` Contains the nmstate NetworkManager plugin, currently for 
+- `./rust/src/lib/nm/` Contains the nmstate NetworkManager plugin, currently for 
 querying network status and applying network states.
```
Comment (ffmancera) [inline CONTRIBUTING.md]: NetworkManager is not a plugin anymore. A better definition would be `Contains the code related to NetworkManager communication and structures.`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -24,20 +24,18 @@ serving the tests of Nmstate.
...
-- `./libnmstate/nispor/` Contains the nmstate nispor plugin, currently for 
+- `./rust/src/lib/nispor/` Contains the nmstate nispor plugin, currently for 
 querying kernel network status. 
 
-- `./libnmstate/nm/` Contains the nmstate NetworkManager plugin, currently for 
+- `./rust/src/lib/nm/` Contains the nmstate NetworkManager plugin, currently for 
 querying network status and applying network states. 
 
-- `./libnmstate/plugins/` Contains the plugins supported by default (e.g. ovsdb). 
+- `./rust/src/lib/ovsdb/` Contains ovsdb plugin supported by default.
```
Comment (ffmancera) [inline CONTRIBUTING.md]: Same here. We do not have plugins anymore. A better definition would be `Contains the code related to ovsdb communication and structures`

**Comment by elomscansio** [comment]
I have modified the CONTRIBUTE.md to include new structure

**Review by ffmancera — COMMENTED** [review]
There are still missing directories, please add them too.

In addition, remember to sign your commit with `-s`

File: CONTRIBUTING.md (hunk 1 above)
Comment (elomscansio) [reply CONTRIBUTING.md]: ok, Thanks

File: CONTRIBUTING.md (hunk 2 above)
Comment (elomscansio) [reply CONTRIBUTING.md]: ok

Comment (elomscansio) [reply CONTRIBUTING.md]: changed

File: CONTRIBUTING.md (hunk 1 above)
Comment (elomscansio) [reply CONTRIBUTING.md]: done

**Comment by ffmancera** [comment]
`There are still missing directories, please add them too.`

This comment was not addressed.

**Comment by elomscansio** [comment]
pls can i get a hint desc of the following remaining dir so i can add them to the project structure section of the [CONTRIBUTE.md](https://github.com/nmstate/nmstate/pull/2597/commits/e934c85ff07fcd2c0257745072a6c9c0faecd643) file

```
./logo - Project logos
./k8s - 
./rust/src/clib - Contain Nmstate C binding files
./rust/src/go/nmstate - 
./rust/src/python/libnmstate - 
./rust/src/python - 
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -24,20 +24,17 @@ serving the tests of Nmstate.
...
+- `./rust/src/lib/ifaces/` Contains the classes including the attributes of the 
 interfaces.
  
-- `./libnmstate/nispor/` Contains the nmstate nispor plugin, currently for 
+- `./rust/src/lib/nispor/` Contains the nmstate nispor plugin, currently for 
 querying kernel network status. 
 
-- `./libnmstate/nm/` Contains the nmstate NetworkManager plugin, currently for 
-querying network status and applying network states. 
+- `./rust/src/lib/nm/` Contains the code related to NetworkManager communication and structures.
```
Comment (liangwen12year) [inline CONTRIBUTING.md]: Contains the code related to applying the settings via NetworkManager backend?

**Comment by cathay4t** [comment]
> pls can i get a hint desc of the following remaining dir so i can add them to the project structure section of the [CONTRIBUTE.md](https://github.com/nmstate/nmstate/pull/2597/commits/e934c85ff07fcd2c0257745072a6c9c0faecd643) file
> 

> ./logo - Project logos

Logos used for publication.

> ./k8s - 

k8s folder holds the scripts for kubernetes-nmstate CI

> ./rust/src/clib - Contain Nmstate C binding files

C binding of nmstate written in rust

> ./rust/src/go/nmstate - 

Go binding of nmstate written in go wrapping the C binding of nmstate

> ./rust/src/python/libnmstate - 
> ./rust/src/python - 

Python library of nmstate written in python wrapping the C binding of nmstate

**Comment by elomscansio** [comment]
> > pls can i get a hint desc of the following remaining dir so i can add them to the project structure section of the [CONTRIBUTE.md](https://github.com/nmstate/nmstate/pull/2597/commits/e934c85ff07fcd2c0257745072a6c9c0faecd643) file
> 
> > ./logo - Project logos
> 
> Logos used for publication.
> 
> > ./k8s -
> 
> k8s folder holds the scripts for kubernetes-nmstate CI
> 
> > ./rust/src/clib - Contain Nmstate C binding files
> 
> C binding of nmstate written in rust
> 
> > ./rust/src/go/nmstate -
> 
> Go binding of nmstate written in go wrapping the C binding of nmstate
> 
> > ./rust/src/python/libnmstate -
> > ./rust/src/python -
> 
> Python library of nmstate written in python wrapping the C binding of nmstate

Thanks 

how about 
./rust/src/lib/policy - Contains code related to enforcing network policy 
./rust/src/lib/statistic, ./rust/src/lib/statistic/feature - Contains code for statistic
./rust/src/lib/query_apply - Contains code that applies to commands
./rust/src/lib/revert - 
./rust/src/lib/unit_tests - Contains test code for lib

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -24,20 +24,28 @@ serving the tests of Nmstate.
...
+- `./rust/src/go/nmstate` Contains the classes including the attributes of the 
 interfaces.
  
-- `./libnmstate/nispor/` Contains the nmstate nispor plugin, currently for 
-querying kernel network status. 
+- `./rust/src/python/libnmstate`, `./rust/src/python` Python library of nmstate written in python wrapping the C binding of nmstate
 
-- `./libnmstate/nm/` Contains the nmstate NetworkManager plugin, currently for 
-querying network status and applying network states. 
+ - `./rust/src/lib/ifaces/` Contains the classes including the attributes of the
```
Comment (cathay4t) [inline CONTRIBUTING.md]: This one is not required

**Review by cathay4t — CHANGES_REQUESTED** [review]
Please remove `./rust/src/lib/ifaces/`.

Besides that looks good to me.

File: CONTRIBUTING.md (hunk 4 above)
Comment (elomscansio) [reply CONTRIBUTING.md]: ok, I have removed it

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -24,20 +24,25 @@ serving the tests of Nmstate.
...
 
-- `./libnmstate/ifaces/` Contains the classes including the attributes of the 
+- `./logo` Logos used for publication.
+
+- `./k8s` k8s folder holds the scripts for kubernetes-nmstate CI
+
+- `./rust/src/clib` C binding of nmstate written in rust
+
+- `./rust/src/go/nmstate` Contains the classes including the attributes of the 
 interfaces.
```
Comment (ffmancera) [inline CONTRIBUTING.md]: This should be `Go library of nmstate wrapping the C bindings`

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -24,20 +24,25 @@ serving the tests of Nmstate.
 
 - `./examples/` - Contains YAML examples for different configurations.
 
-- `./libnmstate/ifaces/` Contains the classes including the attributes of the 
+- `./logo` Logos used for publication.
+
+- `./k8s` k8s folder holds the scripts for kubernetes-nmstate CI
+
+- `./rust/src/clib` C binding of nmstate written in rust
```
Comment (ffmancera) [inline CONTRIBUTING.md]: `C bindings of nmstate` is enough.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -24,20 +24,25 @@ serving the tests of Nmstate.
...
+
+- `./rust/src/go/nmstate` Contains the classes including the attributes of the 
 interfaces.
  
-- `./libnmstate/nispor/` Contains the nmstate nispor plugin, currently for 
+- `./rust/src/python/libnmstate`, `./rust/src/python` Python library of nmstate written in python wrapping the C binding of nmstate
```
Comment (ffmancera) [inline CONTRIBUTING.md]: `rust/src/python Python library of nmstate wrapping the C bindings`

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -24,20 +24,25 @@ serving the tests of Nmstate.
...
+ 
+- `./rust/src/lib/nispor/` Contains the nmstate nispor plugin, currently for
```
Comment (ffmancera) [inline CONTRIBUTING.md]: Plugins do not exist anymore. `Contains the code related to querying the network state with via Nispor`

**Review by ffmancera — DISMISSED** [review]

**Comment by elomscansio** [comment]
I have made the requested changes

**Review by cathay4t — APPROVED** [review]

**Pull request merged**